src-plugins/itkFilters/itkFiltersComponentSizeThresholdProcess_p.h

```diff
@@ -38,32 +38,33 @@ class itkFiltersComponentSizeThresholdProcessPrivate : public itkFiltersProcessB
     
     double minimumSize;
 
-    template <class PixelType> void castToUInt3 ( void )
+    template <class PixelType> int castToUInt3 ( void )
     {
+        //we will later label the image so we don't care about precision.
+        typedef itk::Image< PixelType, 3 > InputImageType;
+        typedef itk::Image< unsigned int, 3 > OutputImageType;
+        typedef itk::CastImageFilter< InputImageType, OutputImageType > CastFilterType;
+
+        typename CastFilterType::Pointer  caster = CastFilterType::New();
+        typename InputImageType::Pointer im = dynamic_cast< InputImageType*>((itk::Object*)(input->data()));
+        caster->SetInput(im);
         try
         {
-            //we will later label the image so we don't care about precision.
-            typedef itk::Image< PixelType, 3 > InputImageType;
-            typedef itk::Image< unsigned int, 3 > OutputImageType;
-            typedef itk::CastImageFilter< InputImageType, OutputImageType > CastFilterType;
-
-            typename CastFilterType::Pointer  caster = CastFilterType::New();
-            typename InputImageType::Pointer im = dynamic_cast< InputImageType*>((itk::Object*)(input->data()));
-            caster->SetInput(im);
             caster->Update();
-            input = medAbstractDataFactory::instance()->createSmartPointer ( "itkDataImageUInt3" );
-            output = medAbstractDataFactory::instance()->createSmartPointer ( "itkDataImageUInt3" );
-            input->setData(caster->GetOutput());
         }
         catch( itk::ExceptionObject & err )
         {
-            std::cerr << "ExceptionObject caught in sizeThresholdingProcess!" << std::endl;
-            std::cerr << err << std::endl;
+            qDebug() << "ExceptionObject caught in sizeThresholdingProcess::castToUInt3 !" << err.GetDescription();
+            return DTK_FAILURE;
         }
+        input = medAbstractDataFactory::instance()->createSmartPointer ( "itkDataImageUInt3" );
+        output = medAbstractDataFactory::instance()->createSmartPointer ( "itkDataImageUInt3" );
+        input->setData(caster->GetOutput());
 
+        return DTK_SUCCEED;
     }
     
-    template <class PixelType> void update ( void )
+    template <class PixelType> int update ( void )
     {
         try
         {
```

Comment:
I would try/catch every filter.
As it is, the dev willing to debug would have to do that anyway.